reco_utils/recommender/deeprec/deeprec_utils.py

```diff
@@ -553,16 +554,26 @@ def cal_metric(labels, preds, metrics):
             f1 = f1_score(np.asarray(labels), pred)
             res["f1"] = round(f1, 4)
         elif metric == "mean_mrr":
-            mean_mrr = np.mean(mrr_score(labels, preds))
+            mean_mrr = np.mean(
+                [   
+                    mrr_score(each_labels, each_preds)
+                    for each_labels, each_preds in zip(labels, preds)
+                ]
+            )
             res["mean_mrr"] = round(mean_mrr, 4)
         elif metric.startswith("ndcg"):  # format like:  ndcg@2;4;6;8
             ndcg_list = [1, 2]
             ks = metric.split('@')
             if len(ks) > 1:
                 ndcg_list = [int(token) for token in ks[1].split(';')]
             for k in ndcg_list:
-                ndcg_temp = np.mean(ndcg_score(labels, preds, k))
-                res["ndcg@{0}".format(k)] = round(ndcg_temp, 4)
+                ndcg_temp = np.mean(
```

Comment:
Why the change of this? `labels` and `preds` themselves are lists so `ndcg_score` directly calculate the metrics for them?

Reply:
I communicate with Jianxun, and make this change. In the old version, the labels and preds are 2-dim arrays, so every input label array and pred array of different impressions should have items of the same numbers. Current implementation support to compute mrr_scores/ndcg scores of different impressions that have different items. The outputs of both versions are mean values of mrr/ndcg scores.